src/server.ts

```diff
@@ -369,38 +398,25 @@ export class SubscriptionServer {
                 throw new Error(error);
               }
 
-              // NOTE: This is a temporary code to identify if the request is a real subscription or only a query/mutation.
-              // As soon as we completely drop the subscription manager and starts handling complete function
-              // from the observable, calling the callback function with (null, null), this can be replaced
-              const isSubscription = this.isASubscriptionRequest(params.query, params.operationName);
-
-              // Create a callback
-              // Error could be a runtime exception or an object with errors
-              // Result is a GraphQL ExecutionResult, which has an optional errors property
-              params.callback = (error: any, result: any) => {
-                if (connectionContext.requests[reqId]) {
-                  if (result) {
-                    this.sendMessage(connectionContext, reqId, MessageTypes.GQL_DATA, result);
-                  }
-
-                  if (error) {
-                    const errorsToSend = error.errors ?
-                      { errors: error.errors } :
-                      { errors: [ { message: error.message } ]};
-                    this.sendMessage(connectionContext, reqId, MessageTypes.GQL_DATA, errorsToSend);
-                  }
-                }
-
-                // NOTE: This is a temporary code to identify if the request is a real subscription or only a query/mutation.
-                // As soon as we completely drop the subscription manager and starts handling complete function
-                // from the observable, calling the callback function with (null, null), this can be replaced
-                if (!isSubscription) {
-                  this.unsubscribe(connectionContext, reqId);
-                  this.sendMessage(connectionContext, reqId, MessageTypes.GQL_COMPLETE, null);
-                }
-              };
-
-              return this.subscribe(connectionContext, reqId, params, isSubscription);
+              return this.execute(this.schema,
+                parse(baseParams.query),
+                this.rootValue,
+                baseParams.context,
+                baseParams.variables,
+                baseParams.operationName)
+              .subscribe({
+                  next: (v: ExecutionResult) => {
+                    // TODO: Handle formatResponse;
```

Comment:
You need them from subscription manager..
The original arch was wrong, they shouldnt be there but post execution..